### DISCUSSION TITLE
Alpine js doesn't detect date property updated via setDate method.

### DISCUSSION THREAD

**Discussion by Shashank-Bhatt1** [alpinejs/alpine#3514] [7. Bugs (archived)] [answered]
Alpine property bounded to date object isn't updated in dom when it was changed via setDate or any other date update methods.

```
<div x-data="{
    currentDate: new Date(),
    setNextWeekDate() {
      this.currentDate.setDate(this.currentDate.getDate() + 7);
      console.log(this.currentDate.toString());
  },
}">
  <span x-text="currentDate"></span>

  <button type="button" @click="setNextWeekDate()">Set next week date</button>
</div>
```

I have created fiddle to highlight this. If we open this and click on the next week button, currentDate is logged to console but isn't updated in dom.

https://jsfiddle.net/3801b5jh/

As a work around, I have added one temporary date, updated that date and then reassign to the currentDate and then it is working fine,

```
 setNextWeekDate() {
     let nextDate = new Date(this.currentDate);
      nextDate.setDate(this.currentDate.getDate() + 7);
     this.currentDate = nextDate;
      console.log(this.currentDate.toString());
  },
```

**Comment by ekwoka** [comment]
Nothing that can be done about that.

Alpine can only respond to certain kinds of changes, and that just isn't one of them. It happens internally.

**Reply by Shashank-Bhatt1** [reply]
@ekwoka Can you explain how alpine changes the property and how one can say that something will react or stay in sync?

Infact if date change is even logged to the console, how/why it isn't updated in dom?

Are you saying you that only certain type of properties will get updated in dom?

**Reply by ekwoka** [reply, marked as answer]
It's Proxies.

So something needs to go and actually set a value to something

```js
obj.property = thing
```

To modify a date, you call a function. Which isn't the same thing. There is no way for a proxy to be aware that anything has changed.

**Comment by Tim-Wils** [comment]
I think the setter in the proxy wrapper (for reactiveness) has to be triggered. This should work: ```this.currentDate = this.currentDate.setDate(this.currentDate.getDate() + 7);```

**Reply by ekwoka** [reply]
`setDate` does not return an updated date value. it returns the UNIX timestamp of the date.
